src/bandit_model.cpp

```diff
@@ -72,27 +241,29 @@ void to_json(nlohmann::json& j, const BanditConfiguration& bc) {
                        {"updatedAt", formatISOTimestamp(bc.updatedAt)}};
 }
 
-void from_json(const nlohmann::json& j, BanditConfiguration& bc) {
-    j.at("banditKey").get_to(bc.banditKey);
-    j.at("modelName").get_to(bc.modelName);
-    j.at("modelVersion").get_to(bc.modelVersion);
-    j.at("modelData").get_to(bc.modelData);
-
-    if (j.contains("updatedAt") && j["updatedAt"].is_string()) {
-        bc.updatedAt = parseISOTimestamp(j["updatedAt"].get<std::string>());
-    }
-}
-
 // BanditResponse serialization
 void to_json(nlohmann::json& j, const BanditResponse& br) {
     j = nlohmann::json{{"bandits", br.bandits}, {"updatedAt", formatISOTimestamp(br.updatedAt)}};
 }
 
 void from_json(const nlohmann::json& j, BanditResponse& br) {
-    j.at("bandits").get_to(br.bandits);
+    br.bandits.clear();
+
+    // Parse bandits - each bandit independently, skip invalid ones
+    if (j.contains("bandits") && j["bandits"].is_object()) {
+        for (auto& [banditKey, banditJson] : j["bandits"].items()) {
+            std::string parseError;
+            auto bandit = internal::parseBanditConfiguration(banditJson, parseError);
+
+            if (bandit) {
+                br.bandits[banditKey] = *bandit;
+            }
+            // If parsing failed, bandit is simply not included in the map
```

Comment:
Similar to another comment -- we should be giving some kind of feedback that there was a problem. Logging to the ApplicationLogger seems like the right approach.